console/src/components/UIKit/atoms/Spinner/index.tsx

```diff
@@ -37,7 +37,11 @@ const spinnerChildStyles = css`
   }
 `;
 
-export const Spinner = props => {
+export type SpinnerProps = {
+  size: string
+};
+
+export const Spinner: React.FC<SpinnerProps> = props => {
```

Comment:
> I see some type errors in declarations.d.ts file after extending HTMLAttributes with css prop. Could you look at this?

I fixed declarations.d.ts error on this commit.

https://github.com/hasura/graphql-engine/pull/4504/commits/5c15555cb69effbeaa1f3d224699c4bdb5e60cb6

> We also need to say that all StyledSpinner props are allowed to be passed. So you can export StyledSpinner's props from Spinner.ts and do:

Your suggestion to fix it caused a type error, so we are dealing with it in a different way.

<details><summary> The error is this. </summary>

<img width="814" alt="スクリーンショット 2020-04-24 6 02 55" src="https://user-images.githubusercontent.com/11070996/80149235-42a5dc00-85f1-11ea-9db2-09a453a0b27e.png">


```
TS2769: No overload matches this call.   Overload 1 of 2, '(props: Pick<Pick<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "style" | "title" | ... 253 more ... | "onTransitionEndCapture"> & ... 6 more ... & ShadowProps<...>, "p" | ... 334 more ... | "textShadow"> & Partial<...>, "p" | ... 334 more ... | "textShadow"> & { ...; } & { ...; }): ReactElement<...>', gave the following error.     Type '{ children: Element[]; height: number; width: number; size: string | (string & (string | number | null)[]) | (string & { [x: string]: string | number | undefined; [x: number]: string | number | undefined; }); ... 76 more ...; textShadow?: string | ... 4 more ... | undefined; }' is not assignable to type 'Pick<Pick<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "style" | "title" | ... 253 more ... | "onTransitionEndCapture"> & ... 6 more ... & ShadowProps<...>, "p" | ... 334 more ... | "textShadow"> & Partial<...>, "p" | ... 334 more ... | "textShadow">'.       Types of property 'color' are incompatible.         Type 'string | number | symbol | (string | number | symbol | null)[] | { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; } | null | undefined' is not assignable to type 'string | (string & (string | number | symbol | null)[]) | (string & { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; }) | undefined'.           Type 'null' is not assignable to type 'string | (string & (string | number | symbol | null)[]) | (string & { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; }) | undefined'.   Overload 2 of 2, '(props: StyledComponentPropsWithAs<"div", any, StyledSpinnerProps, never>): ReactElement<StyledComponentPropsWithAs<"div", any, StyledSpinnerProps, never>, string | ... 1 more ... | (new (props: any) => Component<...>)>', gave the following error.     Type '{ children: Element[]; height: number; width: number; size: string | (string & (string | number | null)[]) | (string & { [x: string]: string | number | undefined; [x: number]: string | number | undefined; }); ... 76 more ...; textShadow?: string | ... 4 more ... | undefined; }' is not assignable to type 'Pick<Pick<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "style" | "title" | ... 253 more ... | "onTransitionEndCapture"> & ... 6 more ... & ShadowProps<...>, "p" | ... 334 more ... | "textShadow"> & Partial<...>, "p" | ... 334 more ... | "textShadow">'.       Types of property 'color' are incompatible.         Type 'string | number | symbol | (string | number | symbol | null)[] | { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; } | null | undefined' is not assignable to type 'string | (string & (string | number | symbol | null)[]) | (string & { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; }) | undefined'.           Type 'null' is not assignable to type 'string | (string & (string | number | symbol | null)[]) | (string & { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; }) | undefined'.
```

</details>

But I'm not sure this is the right way to do it because I'm using a type cast.
We'd love your help.